guardian/prosemirror-elements
# Add image select field

## What does this change?

This PR adds a new image selector field to the existing custom field. The image selector allows a user to pick an image from the Grid and retain the necessary information within the field node's attributes. 

This change adds a `Props` type parameter to the custom field view, allowing a consumer to specify additional properties that are required for rendering the field correctly. In this case, it's used to pass the field image select and crop callback methods which trigger an iframe and provide a function to call with the chosen image data.

The iframe behaviour is facsimile of the `flexible-content` image selection integration and provides broadly the same functionality although implemented a bit differently. This has also meant we need to use dex-nginx so the panda auth will work correctly. The new URL is https://prosemirror-elements.local.dev-gutools.co.uk/. 

## How to test

Load a new instance of the element, press the  "choose image" button, select an image and image crop, observe the image is now attach to the element.

## How can we measure success?

We should functionally be able to use the image-selector field in composer. 

## Images

![image_selector_2](https://user-images.githubusercontent.com/4633246/123928757-1967a500-d986-11eb-921f-64eb9536484b.gif)
